src/app/core/services/data.service.ts

```diff
@@ -15,10 +16,207 @@ export class DataService {
   constructor(private githubService: GithubService) {}
 
   getDataFile(): Observable<{}> {
-    return this.githubService.fetchDataFile().pipe(map((jsonData: {}) => {
-      this.dataFile = <DataFile>{teamStructure: this.extractTeamStructure(jsonData)};
-      return jsonData;
-    }));
+    return this.githubService.fetchDataFile().pipe(
+      map((allCsvDataWrapper: {}) => {
+        return this.constructData(allCsvDataWrapper);
+      }),
+      map((jsonData: {}) => {
+        this.dataFile = <DataFile>{
+          teamStructure: this.extractTeamStructure(jsonData)};
+        return jsonData;
+      })
+    );
+  }
+
+  /**
+   * Merges all parsed Csv Data into a single readable JSON
+   * format.
+   * @param allCsvDataWrapper - Object containing strings of csv data.
+   * @return jsonData - Object representing merged data file.
+   */
+  constructData(allCsvDataWrapper: {}): {} {
+    const jsonData: {} = {};
+    const allCsvData: string = allCsvDataWrapper['data'];
+
+    jsonData['roles'] = this.parseRolesData(allCsvData);
+    jsonData['team-structure'] = this.parseTeamStructureData(allCsvData);
+    jsonData['students-allocation'] = this.parseStudentAllocation(allCsvData);
+    jsonData['tutors-allocation'] = this.parseTutorAllocation(allCsvData);
+    jsonData['admins-allocation'] = this.parseAdminAllocation(allCsvData);
+
+    return jsonData;
+  }
+
+  /**
+   * Parses the input string containing admin allocation information
+   * into application readable Object.
+   * @param csvInput - string containing csv data.
+   * @return admins - object that represents parsed csv data.
+   */
+  parseAdminAllocation(csvInput: string): {} {
+    // CSV Headers
+    const NAME = 'name';
+    const ROLE = 'role';
+
+    const admins = {};
+    let parsedCSV: [{}];
+    parsedCSV = this.csvParser(csvInput);
+
+    // Formats the parsed information for easier app reading
+    parsedCSV.forEach(entry => {
+      if (entry[ROLE] === UserRole.Admin.toLowerCase()) {
+        admins[entry[NAME]] = {};
+      }
+    });
+
+    return admins;
+  }
+
+  /**
+   * Parses the input string containing tutor allocation information
+   * into application readable Object.
+   * @param csvInput - string containing csv data.
+   * @return admins - object that represents parsed csv data.
+   */
+  parseTutorAllocation(csvInput: string): {} {
+    // CSV Headers
+    const NAME = 'name';
+    const TEAM = 'team';
+    const ROLE = 'role';
+
+    const tutors = {};
+    let parsedCSV: [{}];
+    parsedCSV = this.csvParser(csvInput);
+
+    // Formats the parsed information for easier app reading
+    parsedCSV.forEach(entry => {
+      if (entry[ROLE] === UserRole.Tutor.toLowerCase()) {
```

Comment:
Use the guard clause to reduce nesting? If yes, apply in other places too.

Reply:
Yes, Modified in two places.

Reply:
We still have nested code blocks. Would this work?
```js
if !(entry[ROLE] === UserRole.Tutor.toLowerCase()) {
  continue;
}
const tutor = entry[NAME] in tutors ? tutors[entry[NAME]] : {};
tutor[entry[TEAM]] = 'true';
tutors[entry[NAME]] = tutor;
```